ConsumerApi.Tests.Integration/StepDefinitions/DevicesStepDefinitions.cs

```diff
@@ -133,6 +180,13 @@ public async Task ThenD2IsDeleted()
         response.Content.Result!.First().Id?.StringValue.Should().NotBe(_deviceIdD2);
     }
 
+    [Then(@"d3 is not deleted")]
+    public async Task ThenD3IsNotDeleted()
+    {
+        var response = await ListDevices();
+        response.Content.Result!.Where(d => d.Id!.StringValue == _deviceIdD3).Should().NotBeEmpty();
```

Comment:
There is a fluent assertion method for checking the existence of an element via a lambda